### ISSUE TITLE
SyncConfiguration should expose schemaVersion

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-java#3982]
Cocoa exposes `SyncConfiguration.schemaVersion()`. We should do the same. 
The argument is that we want to catch the binding making unintentional changes.

However due to the unique working of sync there is a few caveats and corner cases that needs to be considered. At this point it is unclear how much of this is handled by the Object Store

1) `schemaVersion()` should be considered as `X or bigger`, so calling `realm.getVersion()` might return a number larger than X. It should never return a lower version.

Consider e.g. the following problematic case: 

1) Device A installs schemaVersion:1
2) Device B installs schemaVersion:3
3) Device A installs schemaVersion:2.

If 2/3 both are offline and A comes online later it will write 2 in the metarealm which due to syncs will cause B to suddenly have a lower version number. While not breaking, it will probably be weird for users to experience this. 

Basically we need a `MAX(verA, verB)` as conflict resolution for the metadata schema version number, which I'm not entirely sure how we can do without major refactoring.

**Comment by cmelchior** [comment]
After internal discussion:
- It is possible for the schemaVersion to go down due to merges from other client (although very unlikely). That isn't a problem pr. se, although end users might find it a bit weird. We should just document this for now.
- OS now does require a schema version bump in case of model changes that will cause a migration. This was the primary reason for having it in the first place, which is great.

**Comment by bdash** [comment]
> Cocoa exposes SyncConfiguration.schemaVersion().

It doesn't.

**Comment by cmelchior** [comment]
Sorry, that was my poor description of it. Swift uses `let config = Realm.Configuration(syncConfiguration: SyncConfiguration(user: user, realmURL: syncServerURL))` which by extension mean that the normal means of setting the schema version is accessible.

Java uses two different Builders for `SyncConfiguration` and `RealmConfiguration`, so we need to explicitly add `schemaVersion` to the `SyncConfiguration.Builder`.

**Comment by kneth** [comment]
We need to be able to set the schema version as part of `SyncConfiguration` but we don't have to care if the version of the schema differs when a Realm instance has been created (i.e., simply discard @cmelchior's on `MAX()` above).
